dvc/remote/base.py

```diff
@@ -820,9 +825,102 @@ def cache_exists(self, checksums, jobs=None, name=None):
         Returns:
             A list with checksums that were found in the remote
         """
-        if not self.no_traverse:
-            return list(set(checksums) & set(self.all()))
 
+        if self.no_traverse:
+            return self._cache_exists_no_traverse(checksums, jobs, name)
```

Comment:
Continuing question from https://github.com/iterative/dvc/pull/3501/files#r396638412 . Should we make autodetection default behaviour (no_traverse not set in config) with an ability to explicitly set no_traverse to true/false? Currently it seems like you only do auto-detection when `no_traverse` is `false`, is there a reason for that?

Reply:
This is for the same reason as the other question, assuming that we keep the option so users can force the `no_traverse = true` behavior, then we can skip the autodetection in that case.

Even if the option is explicitly set to `false` by a user, I think it makes sense to still do the autodetection, since in this case the cache content used to estimate the remote size will always be fetched regardless of whether or not we are doing the autodetection.

Reply:
should we consider making a new option if we want to keep the explicit behavior support? It will "reset" existing repos to the desirable auto detection, make this option name "better" (`no_traverse` is very confusing).

Reply:
@pmrowla Forcing `no_traverse=true` doesn't seem like something actual users might use, as usefulness will depend on the number of hashes you need to check. So sounds like that might only be useful for debugging purposes, but in that case forcing `false` also makes sense. But I'm not sure if debugging purposes make sense here at all, to be honest. Maybe we could simply get rid of this config option (just make it obsolete and defunct)? Or do you see any use for it from user or dev perspective?

Reply:
In that case I think it would make the most sense to just drop the config option entirely

Reply:
@pmrowla :+1: Let's do that then in this PR. See `protected` config option as an example of obsoleting.

Reply:
`no_traverse` config option obsoleted in 88ea4dd